Lib/fontTools/cffLib/specializer.py

```diff
@@ -421,10 +421,11 @@ def specializeCommands(commands,
 					(op == commands[i-1][0]) and
 					(not isinstance(args[0], list))):
 				_, other_args = commands[i-1]
-				assert len(args) == 1 and len(other_args) == 1
-				commands[i-1] = (op, [other_args[0]+args[0]])
-				del commands[i]
-				continue
+				if not isinstance(other_args[0], list):
```

Comment:
Combine the if with the conditions before it...

Actually, if both are list, should we add them up?  I think so.

Reply:
And if one is a list and other is a scalar, should the scalar be added to the first item of the list, or all of them?  Depends on whether the list is all master values or master+deltas.  I'm fine skipping that part.

Reply:
Just to make sure that I read your comments correctly:
I think you agree that blend operators should not be represented directly in the charstring.command list. Instead, each blend operator and its args will be replaced by a series of tuples, one for each of the blended args. leaving the expected number of args on the stack for the following marking operator. This makes sense to me, as the current intent of a command list item is that it be self-contained: an operator name, and a list of all its arguments, which is not the case if the blend operator and its args are represented directly. The design cost of the proposed change is that when the program is converted to a command list, the number of regions must be available. For CFF2 Charstrings, this true in the workflows I have seen with the current code base. However, this is true of any logic that needs to make sense of the  blend operator args.

The commandsToProgram does not need a maxstack argument, although it does need the value in order to avoid overrunning the stack, because maxstack for a CFF2 Charstring is now a constant that can be imported from the cffLib. 

And as you suggest, since blend args are all source font values, merging consecutive operations by adding their args is actually pretty simple for any mix of blend args and regular coordinate value args.

Reply:
> And as you suggest, since blend args are all source font values, merging consecutive operations by adding their args is actually pretty simple for any mix of blend args and regular coordinate value args.

Aren't blend args normally deltas?  Those need different treatment adding with a scalar; just the first one should be adjusted.  Right?

Reply:
The rest sounds good.